standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -3585,31 +3585,35 @@ public GetTablesResult get_table_objects_by_name_req(GetTablesRequest req) throw
       if (dbName == null || dbName.isEmpty()) {
         throw new UnknownDBException("DB name is null or empty");
       }
-      if (tableNames == null) {
-        throw new InvalidOperationException(dbName + " cannot find null tables");
-      }
-
-      // The list of table names could contain duplicates. RawStore.getTableObjectsByName()
-      // only guarantees returning no duplicate table objects in one batch. If we need
-      // to break into multiple batches, remove duplicates first.
-      List<String> distinctTableNames = tableNames;
-      if (distinctTableNames.size() > tableBatchSize) {
-        List<String> lowercaseTableNames = new ArrayList<>();
-        for (String tableName : tableNames) {
-          lowercaseTableNames.add(org.apache.hadoop.hive.metastore.utils.StringUtils.normalizeIdentifier(tableName));
+      RawStore ms = getMS();
+      if(tablePattern != null){
```

Comment:
This if condition seems unnecessary now given that we support both tableNames and tablePattern if provided. Just the code in the else block seems sufficient. No?

Reply:
Then we'll not be addressing the scenario of table list can be empty and table pattern is set to some regex. We'll we returning an empty list in this case